File: src/core/validators/SharedProperty/SharedDecimalPropertyTypeMustMatchACommonSimpleDecimal.cs
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.SharedProperty
{
    public class SharedDecimalPropertyTypeMustMatchACommonSimpleDecimal : ValidationRuleBase<MetaEdGrammar.SharedDecimalPropertyContext>
    {
        private readonly ISymbolTable _symbolTable;

        public SharedDecimalPropertyTypeMustMatchACommonSimpleDecimal(ISymbolTable symbolTable)
        {
            _symbolTable = symbolTable;
        }

        public override bool IsValid(MetaEdGrammar.SharedDecimalPropertyContext context)
        {
            var identifierToMatch = context.sharedPropertyType().GetText();
            var commonDecimalType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_DECIMAL);

            return _symbolTable.IdentifierExists(commonDecimalType, identifierToMatch);
        }

        public override string GetFailureMessage(MetaEdGrammar.SharedDecimalPropertyContext context)
        {
            return string.Format("Shared property '{0}' does not match any declared common decimal.", context.propertyName().GetText());
        }
    }
}

File: src/core/validators/SharedProperty/SharedDecimalPropertyTypeMustMatchACommonSimpleDecimal.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
import {ISymbolTable} from '../SymbolTable'
export class SharedDecimalPropertyTypeMustMatchACommonSimpleDecimal extends ValidationRuleBase<MetaEdGrammar.SharedDecimalPropertyContext>
{
    private symbolTable: ISymbolTable;
    constructor(symbolTable: ISymbolTable) {
        super();
        this.symbolTable = symbolTable;
    }
    public isValid(context: MetaEdGrammar.SharedDecimalPropertyContext): boolean {
        let identifierToMatch = context.sharedPropertyType().GetText();
        let commonDecimalType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_DECIMAL);
        return this.symbolTable.identifierExists(commonDecimalType, identifierToMatch);
    }
    public getFailureMessage(context: MetaEdGrammar.SharedDecimalPropertyContext): string {
        return `Shared property '${}' does not match any declared common decimal.", context.propertyName().GetText());
    }
}
